Feature/lp162 office view model

AndresIvanMedinaH opened the pull request:
**• Task’s name**
Office - ViewModel   

    **• Task’s link to navigate to jira**
https://devaptivist.atlassian.net/browse/LP-162?atlOrigin=eyJpIjoiOTM5YWVjMmQyZWYxNDQzMDhhZWU2Y2FkNjg3ZmU3YTEiLCJwIjoiaiJ9


updated-->
![imagen](https://github.com/irv205/droid_leaspert/assets/128720218/b9faa140-5d3f-4ebf-8da8-697b32c4a97b)

AndresIvanMedinaH committed:
TO-DO
+Verify that the "Information" title is displayed correctly.
+Verify that the address of the office is displayed correctly.
+Verify that the office hours are displayed correctly.
+Verify that the "Call" button opens the phone app with the office phone number.
+Verify that the "Mail" button opens the email app with the office email.

Solution proposition:
based on SOLID:
Use an Interface to:
Grant permission to whoever implements this interface
On Implementation:
Generate the logic that deal with the permission asking based on the different kind of permissions
this implementation requieres the Context where it is need it or being called

use an Interface:
IintentActions that deals with the different permissions needed by intents along the app

On Implementation:
create the necessary methods to process the launching of the intents and receives the inection of the
permissionProvider
______________________________________________________________
Office ->renamed to -> OfficeView to keep uniformity on naming Views, ViewModels and so
______________________________________________________________
Changes:
Permissions removed from the main manifest
Bundle removed from the gradle
Files removed from the sprint
OfficeInfoItemModel--> renamed to OfficeModel
OfficeView--> Text() components replaced for Text functions defined for each case of text that exists in the app
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________

AndresIvanMedinaH committed:
TO-DO
+Verify that the "Information" title is displayed correctly.
+Verify that the address of the office is displayed correctly.
+Verify that the office hours are displayed correctly.
+Verify that the "Call" button opens the phone app with the office phone number.
+Verify that the "Mail" button opens the email app with the office email.

Solution proposition:
based on SOLID:
Use an Interface to:
Grant permission to whoever implements this interface
On Implementation:
Generate the logic that deal with the permission asking based on the different kind of permissions
this implementation requieres the Context where it is need it or being called

use an Interface:
IintentActions that deals with the different permissions needed by intents along the app

On Implementation:
create the necessary methods to process the launching of the intents and receives the inection of the
permissionProvider
______________________________________________________________
Office ->renamed to -> OfficeView to keep uniformity on naming Views, ViewModels and so
______________________________________________________________
Changes:
Permissions removed from the main manifest
Bundle removed from the gradle
Files removed from the sprint
OfficeInfoItemModel--> renamed to OfficeModel
OfficeView--> Text() components replaced for Text functions defined for each case of text that exists in the app
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________

AndresIvanMedinaH committed:
TO-DO
+Verify that the "Information" title is displayed correctly.
+Verify that the address of the office is displayed correctly.
+Verify that the office hours are displayed correctly.
+Verify that the "Call" button opens the phone app with the office phone number.
+Verify that the "Mail" button opens the email app with the office email.

Solution proposition:
based on SOLID:
Use an Interface to:
Grant permission to whoever implements this interface
On Implementation:
Generate the logic that deal with the permission asking based on the different kind of permissions
this implementation requieres the Context where it is need it or being called

use an Interface:
IintentActions that deals with the different permissions needed by intents along the app

On Implementation:
create the necessary methods to process the launching of the intents and receives the inection of the
permissionProvider
______________________________________________________________
Office ->renamed to -> OfficeView to keep uniformity on naming Views, ViewModels and so
______________________________________________________________
Changes:
Permissions removed from the main manifest
Bundle removed from the gradle
Files removed from the sprint
OfficeInfoItemModel--> renamed to OfficeModel
OfficeView--> Text() components replaced for Text functions defined for each case of text that exists in the app
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel
# Conflicts:
#	feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt

AndresIvanMedinaH committed:
TO-DO
+Verify that the "Information" title is displayed correctly.
+Verify that the address of the office is displayed correctly.
+Verify that the office hours are displayed correctly.
+Verify that the "Call" button opens the phone app with the office phone number.
+Verify that the "Mail" button opens the email app with the office email.

Solution proposition:
based on SOLID:
Use an Interface to:
Grant permission to whoever implements this interface
On Implementation:
Generate the logic that deal with the permission asking based on the different kind of permissions
this implementation requieres the Context where it is need it or being called

use an Interface:
IintentActions that deals with the different permissions needed by intents along the app

On Implementation:
create the necessary methods to process the launching of the intents and receives the inection of the
permissionProvider
______________________________________________________________
Office ->renamed to -> OfficeView to keep uniformity on naming Views, ViewModels and so
______________________________________________________________
Changes:
Permissions removed from the main manifest
Bundle removed from the gradle
Files removed from the sprint
OfficeInfoItemModel--> renamed to OfficeModel
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel
# Conflicts:
#	feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt

AndresIvanMedinaH committed:
TO-DO
+view model implementation

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel
# Conflicts:
#	feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt

AndresIvanMedinaH committed:
TO-DO
+view model implementation

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel
# Conflicts:
#	feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel

AndresIvanMedinaH committed:
TO-DO
+view model implementation

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel
# Conflicts:
#	feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt

AndresIvanMedinaH committed:
TO-DO
+view model implementation

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel
# Conflicts:
#	feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt

AndresIvanMedinaH committed:
TO-DO
+view model implementation

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel
# Conflicts:
#	feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt

AndresIvanMedinaH commented:
in this pr, you might find the implementation of the view model, the refactoring task of the components was delegated into another PR

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel

AndresIvanMedinaH committed:
Merge branch 'develop' into feature/LP162-Office-viewModel

AndresIvanMedinaH committed:
TO-DO
+view model implementation

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel
# Conflicts:
#	feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt

carlosph816 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -1,55 +1,93 @@
+    Column(
+        Modifier
+            .fillMaxSize()
+            .padding(top = 32.dp)) {
```
I think we need to use the Desing Sizes here

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel

AndresIvanMedinaH committed:
TO-DO
+view model implementation

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Files modified:
-OfficeModel  renamed to Office  moved to Domain
-OfficeView
-OfficeViewModel
-Strings.xml added fields

Files removed
-ContactTelephone
-Email

AndresIvanMedinaH commented:
agree, that was placed just to watch the title, talking about the top
padding daniel were adding due to the toolbar heigh size, so that size need
to be gone

El jue, 18 may 2023 a las 12:59, Carlos ***@***.***>)
escribió:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In
> feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt
> <https://github.com/irv205/droid_leaspert/pull/123#discussion_r1198061609>
> :
>
> > + * @param onCallButtonClicked launches the call intent
> + * @param onEmailButtonClicked launches the Email intent
> + * @author Yaritza Moreno
> + * Modified by:
> + * @author Andres Ivan Medina
> + */
> ***@***.***
> +fun OfficeUI(
> +    officeData: OfficeModel,
> +    onCallButtonClicked: () -> Unit,
> +    onEmailButtonClicked: () -> Unit,
> +) {
> +    Column(
> +        Modifier
> +            .fillMaxSize()
> +            .padding(top = 32.dp)) {
>
> I think we need to use the Desing Sizes here
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/irv205/droid_leaspert/pull/123#pullrequestreview-1433087432>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A6WB2WSCQSWGONT7OGWFRRLXGZIOJANCNFSM6AAAAAAYCYAEPU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Lupillo17 commented on core/domain/src/main/java/com/itenum/domain/ContactTelephone.kt:
by the moment move this file to **feature/office**

AndresIvanMedinaH replied:
File was already removed due to conflicts with the sprint

Lupillo17 commented on core/domain/src/main/java/com/itenum/domain/Email.kt:
by the moment move this file to feature/office

AndresIvanMedinaH replied:
File was already removed due to conflicts with the sprint

Lupillo17 commented on core/data/build.gradle:
```diff
@@ -31,6 +31,9 @@ android {
 dependencies {
+    implementation libs.bundles.hilt
+
+    implementation project(path: ':core:domain')
```
sort the implementations, libs with libs and project with project

AndresIvanMedinaH replied:
sorted

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/data/OfficeModel.kt:
```diff
@@ -0,0 +1,19 @@
+ * @param email electronic email of the business
+ * @author Andres Ivan Medina
+ */
+data class OfficeModel(
```
rename data class, if this is a response from external data source, chage it for **OfficeDTO** if is a domain data class, you could name it as **Office** or **OfficeDomain**.

If it is a Domain data class, move to domain layer

AndresIvanMedinaH replied:
renamed

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -1,55 +1,93 @@
-fun OfficeUI(viewModel: OfficeViewModel = hiltViewModel()) {
-    Column(Modifier.fillMaxSize()) {
+fun OfficeView() {
+    val viewModel = OfficeViewModel()
```
pass viewModel as parameter to the OfficeView

AndresIvanMedinaH replied:
this was said before that viewmodel must not be passed as parameter along compose functions  :O

AndresIvanMedinaH replied:
Adding hilt implementation to the viewModel to receive the injection ...

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
Use **ctrl + L** to reformat code

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -1,55 +1,93 @@
+
+    val OfficeData = viewModel.officeInfo
+
+    OfficeData?.let {
```
remove the scope function, instead of scope function use **state** value from **OfficeViewModel** to handle each state.

You can use a **when** to handle each state (loading, success, error).

AndresIvanMedinaH replied:
added the state control inside main view to catch emission of data through

  val uiState = viewModel.state.collectAsState()

    when(uiState.value){
       is OfficeStateResponse.Error ->{ ShowStateContent(stringResource(R.string.Error))}
       is OfficeStateResponse.Loading->{ ShowStateContent(stringResource(R.string.Loading))}
       is OfficeStateResponse.Success->{
            val OfficeData = viewModel.officeInfo

            OfficeData?.let {
                OfficeUI(
                    it,
                    onCallButtonClicked = { viewModel.makeCall() },
                    onEmailButtonClicked = { viewModel.sendEmail() }
                )
            }
        }
    }

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeViewModel.kt:
```diff
@@ -0,0 +1,98 @@
+ *
+ *@author  Yaritza Moreno
+ *
+ *@author Andres Ivan Medina
```
Should be 1 author only, replace for **@modifiedBy**

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeViewModel.kt:
```diff
@@ -0,0 +1,98 @@
+
+    internal fun getOfficeInformation() = viewModelScope.launch {
+        when (officeInfoResponse) {
+            is DataState.Success -> {
```
emmit success state like in error state and loading state

AndresIvanMedinaH replied:
added

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeViewModel.kt:
```diff
@@ -0,0 +1,98 @@
+)
+
+/**
+ * class:OfficeStateList  represents the states of the data
```
modify class name from documentation, is different to the class name below

AndresIvanMedinaH replied:
got it

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -1,55 +1,93 @@
+    onEmailButtonClicked: () -> Unit,
+) {
+    Column(
+        Modifier
```
add named param: modifier=Modifier to **Column**

AndresIvanMedinaH replied:
got it

Lupillo17 commented on feature/office/src/main/res/values/strings.xml:
```diff
@@ -3,6 +3,8 @@
+    <string name="LPHours">Hours:</string>
     <string name="LPAddress">Address: 4950 Gaidrew, Alpharetta, GA, 30022</string>
-    <string name="LPHours">Hours: 9AM - 7PM</string>
+    <string name="LPphoneServiceContact">+1 452 123 4567</string>
```
change the fisrt letter after LP to Uppercase should be LPPhoneServiceContact

AndresIvanMedinaH replied:
got it

Lupillo17 commented on feature/office/src/main/res/values/strings.xml:
```diff
@@ -3,6 +3,8 @@
     <string name="LPAddress">Address: 4950 Gaidrew, Alpharetta, GA, 30022</string>
-    <string name="LPHours">Hours: 9AM - 7PM</string>
+    <string name="LPphoneServiceContact">+1 452 123 4567</string>
+    <string name="LPcontactEmail">contact@leaspert.com</string>
```
change the first letter after LP to Uppercase, should be LPContactEmail

AndresIvanMedinaH replied:
got it

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -87,13 +108,11 @@ fun OfficeUI(
                 textIndent = TextIndent(firstLine = 14.sp, restLine = 3.sp)
-            ),
-            modifier = Modifier.padding(top = sizes.extraSize10)
+            ), modifier = Modifier.padding(top = sizes.extraSize10)
```
Remember pass the modifier as first parameter in each component

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeViewModel.kt:
```diff
@@ -76,23 +65,24 @@ class OfficeViewModel() : ViewModel() {
-    OfficeModel(
+val officeResponse: DataState<Office> = DataState.Success(
+    Office(
         "4950 Gaidrew, Alpharetta, GA, 30022 ",
```
add named param address = ....... 4950 Gaidrew, Alpharetta, GA, 30022. Note: delete extra space at end of the string.
same for schedule = 9AM-6PM

AndresIvanMedinaH committed:
Code corrections 30/05/2023
over reformating actions
over re ordering libaries

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -57,28 +57,27 @@ fun OfficeView(viewModel: OfficeViewModel = hiltViewModel()) {
             OfficeData?.let {
-                OfficeUI(
-                    it,
+                OfficeUI(it,
```
reformat code

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -57,28 +57,27 @@ fun OfficeView(viewModel: OfficeViewModel = hiltViewModel()) {
-    LPTitleLarge(label = response, color = MaterialTheme.colorScheme.tertiary, weight =FontWeight.Bold)
+fun ShowStateContent(response: String) {
+    LPTitleLarge(
+        label = response, color = MaterialTheme.colorScheme.tertiary, weight = FontWeight.Bold
```
reformat code

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -108,15 +109,17 @@ fun OfficeUI(
                 textIndent = TextIndent(firstLine = 14.sp, restLine = 3.sp)
-            ), modifier = Modifier.padding(top = sizes.extraSize10)
+            ),
+            modifier = Modifier.padding(top = sizes.extraSize10)
```
pass modifier as first parameter

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -108,15 +109,17 @@ fun OfficeUI(
         Text(
-            text = officeData.schedule,
+            text = schedule,
             modifier = Modifier.padding(top = sizes.extraSize10),
```
pass parameter modifier as first parameter

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -108,15 +109,17 @@ fun OfficeUI(
 
         Text(
-            text = officeData.schedule,
+            text = schedule,
```
add stringResource of LPHour and pass **schelude** as parameter

AndresIvanMedinaH committed:
Code corrections 30/05/2023
over reformating actions
over re ordering libaries
downgrade build project version of com.android.application and library

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -109,18 +113,16 @@ fun OfficeUI(
         Text(
-            text = address,
-            style = TextStyle(
+            modifier = Modifier.padding(top = sizes.extraSize10), text = address, style = TextStyle(
```
reformat code

AndresIvanMedinaH committed:
Code corrections 30/05/2023
over re formating actions
over re ordering libaries
downgrade build project version of com.android.application and library

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel

AndresIvanMedinaH committed:
Merge branch 'develop' into feature/LP162-Office-viewModel

# Conflicts:
#	feature/office/src/main/java/com/iteneum/office/domain/Office.kt
#	feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt
#	feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeViewModel.kt

AndresIvanMedinaH committed:
Code corrections 30/05/2023
over re formating actions
over re ordering libaries
downgrade build project version of com.android.application and library
